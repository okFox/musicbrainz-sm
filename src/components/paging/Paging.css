button {
    height: 40px;
    width: 80px;
    border-radius: 5px;
    font-size: larger;
    color: white;
    background-color: gray;
    margin-bottom: 25px;
}
span {
    font-size: larger;
    font-family: 'Acme', sans-serif;
    padding: 10px;
}