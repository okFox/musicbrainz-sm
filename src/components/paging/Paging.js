import React from 'react';
import PropTypes from 'prop-types';
import './Paging.css';

const Paging = ({ currentPage, handleNextClick, handlePreviousClick }) => (
  <>
    <button onClick={handlePreviousClick}>&lt;</button>
    <span>Page {currentPage}</span>
    <button onClick={handleNextClick}>&gt;</button>
  </>
);

Paging.propTypes = {
  currentPage: PropTypes.string.isRequired,
  handleNextClick: PropTypes.func.isRequired,
  handlePreviousClick: PropTypes.func.isRequired
};

export default Paging;
