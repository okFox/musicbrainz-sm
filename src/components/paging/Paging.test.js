import React from 'react';
import { shallow } from 'enzyme';
import Paging from './Paging';

describe('Paging component', () => {
  it('renders Paging', () => {
    const wrapper = shallow(<Paging currentPage="1" handleNextClick={()=>{}} handlePreviousClick={()=>{}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
