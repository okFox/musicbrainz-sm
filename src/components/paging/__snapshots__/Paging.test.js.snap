// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component renders Paging 1`] = `
<Fragment>
  <button
    onClick={[Function]}
  >
    &lt;
  </button>
  <span>
    Page 
    1
  </span>
  <button
    onClick={[Function]}
  >
    &gt;
  </button>
</Fragment>
`;
